src/controller/CHIPDevice.cpp

```diff
@@ -97,16 +97,22 @@ CHIP_ERROR Device::Serialize(SerializedDevice & output)
     uint16_t serializedLen = 0;
     SerializableDevice serializable;
 
-    nlSTATIC_ASSERT_PRINT(BASE64_ENCODED_LEN(sizeof(serializable)) <= sizeof(output.inner),
-                          "Size of serializable should be <= size of output");
+    static_assert(BASE64_ENCODED_LEN(sizeof(serializable)) <= sizeof(output.inner),
+                  "Size of serializable should be <= size of output");
 
     CHIP_ZERO_AT(serializable);
 
     memmove(&serializable.mOpsCreds, &mPairing, sizeof(mPairing));
     serializable.mDeviceId   = Encoding::LittleEndian::HostSwap64(mDeviceId);
     serializable.mDevicePort = Encoding::LittleEndian::HostSwap16(mDevicePort);
-    nlSTATIC_ASSERT_PRINT(sizeof(serializable.mDeviceAddr) <= INET6_ADDRSTRLEN,
-                          "Size of device address must fit within INET6_ADDRSTRLEN");
+#if CHIP_SYSTEM_CONFIG_USE_LWIP
```

Comment:
We need some GH issue for supporting LWIP:
  - LWIP is likely our main customer (embedded devices)
  - LWIP has a concept of interface ID (it is a structure though, so we probably want to serialize the name)

We need to make this serialization support both rather than just a 'serialize 64bit number'

Reply:
It seems LWIP name is 2 characters ... so maybe an ugly encode hack to make it a 64bit number works (with a ton of commenting)